### PR TITLE
Axios Setup

### LINKED ISSUE #6: axios setup
Create api services like apiFunction and interceptor functions with required parameters.

🕐  :- 1 to 2 hours

### PR DESCRIPTION
**Pull request by Naveen-gali**
### What does this PR do?

- Integrates Axios to Project.
- Axios Interceptors Setup.
- Get Api Request Addition.

### Related Issues

- closes #6 

### How to test

* Clone the Repository.
* Switch to `dev-axiosSetup` branch.
* Do `yarn` to install the dependencies.
* Do `yarn run build-stage-ios` for `ios` or `yarn run build-stage-android` for `android`.
* Look out console for the api response.

### Checklist

- [x] I have tested this code in both IOS and Android.
- [x] I have tested this code locally and verified that it works as intended.
- [x] I have checked the code for formatting, syntax, and other best practices.
- [x] I have ensured that my changes do not break any existing functionality.
- [ ] I have attached the necessary ScreenShots or Screen Recordings.

#### Links to Know more
- [Axios](https://axios-http.com/docs/intro)
- [Axios Interceptors](https://axios-http.com/docs/interceptors)

### Packages Added (if any)

| Package | Reason |
| ------- | ------ |
| Axios    | To Make Api Calls       |

### CODE REVIEW

File: src/api/api.ts [hunk 1]
```diff
@@ -1 +1,45 @@
...
+  },
+);
+
+export function GetApi<T>(
+  location: string,
+  sort: string,
+  limit: number,
+): Promise<T> {
+  return YelpApi.get(
+    `/search?location=${location}&sort_by=${sort}&limit=${limit}`,
```
Comment (prajnab-mm) [inline src/api/api.ts]: Why is the URL hardcoded?

File: src/api/api.ts [hunk 2]
```diff
@@ -1 +1,45 @@
...
+YelpApi.interceptors.response.use(
+  function (response) {
+    return response.data;
+  },
+  function (error) {
+    return Promise.reject(error);
+  },
+);
+
+export function GetApi<T>(
```
Comment (prajnab-mm) [inline src/api/api.ts]: Create a single function that accepts all HtpMethod Apis - https://github.com/axios/axios#axiosrequestconfig

File: src/api/api.ts [hunk 3]
```diff
@@ -1 +1,45 @@
...
+  limit: number,
+): Promise<T> {
+  return YelpApi.get(
+    `/search?location=${location}&sort_by=${sort}&limit=${limit}`,
+  )
+    .then(response => {
+      return response;
+    })
+    .catch(err => {
+      return err;
```
Comment (prajnab-mm) [inline src/api/api.ts]: Handle common errors and display alert in case screen wants you to display an alert

File: src/api/api.ts [hunk 4]
```diff
@@ -1 +1,45 @@
...
+
+export function GetApi<T>(
+  location: string,
+  sort: string,
+  limit: number,
+): Promise<T> {
+  return YelpApi.get(
+    `/search?location=${location}&sort_by=${sort}&limit=${limit}`,
+  )
+    .then(response => {
```
Comment (prajnab-mm) [inline src/api/api.ts]: Add types for Success and Error response

File: ios/yelpApp.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -84,7 +86,11 @@
 		8AA162D329C41E4C00018445 /* SimpleLineIcons.ttf */ = {isa = PBXFileReference; lastKnownFileType = file; path = SimpleLineIcons.ttf; sourceTree = "<group>"; };
 		8AA162D429C41E4C00018445 /* MaterialIcons.ttf */ = {isa = PBXFileReference; lastKnownFileType = file; path = MaterialIcons.ttf; sourceTree = "<group>"; };
 		8AFEDB8529C42A4A009E0BA9 /* Config.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; path = Config.xcconfig; sourceTree = "<group>"; };
+		8CC1F476BE774552B52214F3 /* Poppins-Regular.ttf */ = {isa = PBXFileReference; explicitFileType = undefined; fileEncoding = undefined; includeInIndex = 0; lastKnownFileType = unknown; name = "Poppins-Regular.ttf"; path = "../src/assets/fonts/src/Poppins-Regular.ttf"; sourceTree = "<group>"; };
```
Comment (prajnab-mm) [inline ios/yelpApp.xcodeproj/project.pbxproj]: Check why its duplicated

File: .env [hunk 6]
```diff
@@ -1 +1,2 @@
-ENV = "ENV"
\ No newline at end of file
+ENV = "ENV"
+API_URL = "https://api.yelp.com/v3/businesses"
```
Comment (prajnab-mm) [inline .env]: versioning and businesses should not be part of base URL

File: src/screens/HomeScreen/HomeScreen.tsx [hunk 7]
```diff
@@ -1,9 +1,40 @@
...
+import {SEARCH_BUSINESS} from '../../constants';
+import {ErrorType, BusinessSuccessType, SortByEnum} from './types';
 
 const HomeScreen = () => {
+  const getBusinesses = (
+    location: string,
+    sort_by: SortByEnum,
+    limit: number,
+  ) => {
+    return Api<BusinessSuccessType | ErrorType>(
```
Comment (prajnab-mm) [inline src/screens/HomeScreen/HomeScreen.tsx:16]: Always call API using a service or model

File: src/screens/HomeScreen/types.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  display_address: string[];
+  state: string;
+  zip_code: string;
+}
+
+export interface Region {
+  center: Center;
+}
+
+export interface Center {
```
Comment (prajnab-mm) [inline src/screens/HomeScreen/types.ts:51]: Duplicate interface as Coordinates

File: src/screens/HomeScreen/types.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
+export interface Root {
```
Comment (prajnab-mm) [inline src/screens/HomeScreen/types.ts:1]: Too generic. Give proper name

File: src/constants/urlConstants.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+const BUSINESS_URL = '/v3/businesses';
+
+const SEARCH_BUSINESS = BUSINESS_URL + '/search';
+
+export {SEARCH_BUSINESS};
```
Comment (prajnab-mm) [inline src/constants/urlConstants.ts:5]: Do not export as individual keys. Instead export as ApiEndpoints. Also, move this to within api folder

**Comment by Naveen-gali** [comment]
As i am not satisfied with the outcome of this pull request i am reworking on the code. Thus i am closing this pull request.

**Pull request closed without merging**